pkg/asset/agent/image/agentimage.go

```diff
@@ -263,14 +270,22 @@ func (a *AgentImage) PersistToFile(directory string) error {
 		if err != nil {
 			return err
 		}
-		logrus.Infof("Generated minimal ISO at %s", agentIsoFile)
+		msg := fmt.Sprintf("Generated minimal ISO at %s.", agentIsoFile)
+		if a.imageExpirsAt != "" {
+			msg = fmt.Sprintf("%s The ISO is valid up to %s", msg, a.imageExpirsAt)
+		}
+		logrus.Info(msg)
 	} else {
 		// Generate full ISO
 		err = isoeditor.Create(agentIsoFile, a.tmpPath, a.volumeID)
 		if err != nil {
 			return err
 		}
-		logrus.Infof("Generated ISO at %s", agentIsoFile)
+		msg := fmt.Sprintf("Generated ISO at %s.", agentIsoFile)
+		if a.imageExpirsAt != "" {
+			msg = fmt.Sprintf("%s The ISO is valid up to %s", msg, a.imageExpirsAt)
+		}
+		logrus.Info(msg)
```

Comment:
Could you do the check of a.imageExpirsAt and the logrus.Info(msg) after the if/else to avoid the code duplication?